File: test_doubles/csharp/TestDoubles/TestDoubles/TestedObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TestDoubles
{
    public class TestedObject
    {
        private IDependency dependency;

        public TestedObject(IDependency dependency) 
        {
            this.dependency = dependency;
        }

        public int ComputeSomething() 
        {
            return 42 * dependency.ComputeAndReturnValue();
        }
    }
}

File: test_doubles/java/TestDoubles/src/main/java/types/TestedObject.java
package types;

public class TestedObject {

    private Collaborator collaborator;

    public TestedObject(Collaborator collaborator) {
        this.collaborator = collaborator;
    }

    public int computeSomething() {
        return 42 * collaborator.computeAndReturnValue();
    }
}
